CHANGELOG.md

```diff
@@ -1,25 +1,223 @@
+
+```javascript
+// Example object schema
+const TaskSchema = {
+  name: "Task",
+  properties: {
+    description: /* property schema (shorthand or object form) */,
+  },
+};
+
+// Explicitness
+"[]"          // Bad (previously parsed as implicit "mixed")
+"mixed[]"     // Good
+
+{ type: "list" }                              // Bad
+{ type: "list", objectType: "mixed" }         // Good
+
+// Mixing shorthand and object form
+{ type: "int[]" }                             // Bad
+"int[]"                                       // Good
+{ type: "list", objectType: "int" }           // Good
+
+{ type: "int?" }                              // Bad
+"int?"                                        // Good
+{ type: "int", optional: true }               // Good
+
+// Specifying object types
+{ type: "SomeType" }                          // Bad
+"SomeType"                                    // Good
+{ type: "object", objectType: "SomeType" }    // Good
+
+{ type: "object[]", objectType: "SomeType" }  // Bad
+"SomeType[]"                                  // Good
+{ type: "list", objectType: "SomeType" }      // Good
+
+{ type: "linkingObjects", objectType: "SomeType", property: "someProperty" } // Good
+```
+
+* To prevent modifying end-users' class-based model classes, we’re now wrapping class-based models in our own model. Objects will still pass `instanceof SomeClass` checks, however, code which is directly using prototype or constructor comparisons will fail:
+  
+```javascript
+Object.getPrototypeOf(object) == CustomObject.prototype // No longer works
+object.constructor == CustomObject // No longer works
+```
+
+* Symbols used to be accepted as keys in a dictionary, where they were coerced to strings prior to performing lookup. This was undocumented behaviour that makes little sense in practice (and arguably defeats the main purpose of the JS `Symbol` type). In the new SDK, using a Symbol as a key in a dictionary will throw.
+* As a part of migrating to NAPI (since ~ v6), we saw no performant way to support getting property names of a Realm.Object via the standard Object.keys(obj). As a side-effect we stopped supporting the object spread operator `{...obj}` and introduced `Realm.Object#keys()`, `Realm.Object#entries()` and `Realm.Object#toJSON()` methods were introduced as a workaround. The new SDK wraps its accessor objects in a Proxy trapping the ownKeys operation which enables calls to the standard `Object.keys(obj)` and the spread operator `{...obj}` to work correctly, with minimal performance impact on normal accesses. Therefore, we are deprecating the APIs with the @deprecation annotation and a `console.warn` when calling RealmObject#keys() and RealmObject#entries(). RealmObject#toJSON still serves the purpose of producing a circularly referencing object graph. We would love the community's feedback on this!
```

Comment:
Since a lot of people has been hit by this, we might want to move this up?

Reply:
Also, we might want to mention the support of the spread operator as an enhancement, new feature or fix.

Reply:
I will move it up and add that the broken spread operator has been fixed (with a link to https://github.com/realm/realm-js/issues/2844).